src/circuit/gadget/utilities.rs

```diff
@@ -97,3 +98,195 @@ pub fn transpose_option_array<T: Copy + std::fmt::Debug, const LEN: usize>(
     }
     ret
 }
+
+/// Subsets a field element to a specified bitrange (little-endian)
```

Comment:
It was unclear to me what this meant without looking at the implementation. I suggest something like:
```suggestion
/// Takes a specified subsequence of the little-endian bit representation of a field element.
```